h/static/scripts/directive/annotation.coffee

```diff
@@ -108,7 +108,12 @@ AnnotationController = [
     ###
     this.authorize = (action) ->
       return false unless model?
-      permissions.permits action, model, auth.user
+      # TODO: this should use auth instead of permissions but we might need
+      # an auth cache or the JWT -> userid decoding might start to be a
+      # performance bottleneck and we would need to get the id token into the
+      # session, which we should probably do anyway (and move to opaque bearer
+      # tokens for the access token).
+      return permissions.permits action, model, session.state.userid
```

Comment:
Not sure I understand this. `auth.user` is computed once on auth change, not on every access.